include/oneapi/dpl/internal/random_impl/linear_congruential_engine.h

```diff
@@ -141,25 +147,54 @@ class linear_congruential_engine
             state_[__i] = mod_scalar(state_[__i - 1u]);
     }
 
+    // Internal function for calculate degrees of multiplier
+    scalar_type
+    pow_mult_n(unsigned long long __num_to_skip)
+    {
+        ::std::uint64_t __a2;
+        ::std::uint64_t __a = multiplier;
+        scalar_type __r;
+
+        __r = 1;
+
+        do
+        {
+            if (__num_to_skip & 1)
+            {
+                __a2 = static_cast<::std::uint64_t>(__r) * static_cast<::std::uint64_t>(__a);
+                __r = __a2 % modulus;
+            }
+
+            __num_to_skip >>= 1;
+            __a2 = static_cast<::std::uint64_t>(__a) * static_cast<::std::uint64_t>(__a);
```

Comment:
static_cast for _a is not needed

Reply:
agree - double checked where static_casts are needed here